Userland/Games/Chess/ChessWidget.cpp

```diff
@@ -837,13 +866,54 @@ int ChessWidget::resign()
     board().set_resigned(m_board.turn());
 
     set_drag_enabled(false);
+    m_timer->stop();
     update();
     auto const msg = Chess::Board::result_to_string(m_board.game_result(), m_board.turn());
     GUI::MessageBox::show(window(), msg, "Game Over"sv, GUI::MessageBox::Type::Information);
 
     return 0;
 }
 
+void ChessWidget::initialize_timer()
+{
+    m_timer = Core::Timer::create_repeating(
+        1000, [this] {
+            auto white_time = m_time_control_seconds - m_white_time_elapsed;
+            auto black_time = m_time_control_seconds - m_black_time_elapsed;
+            if (m_board.turn() == Chess::Color::White) {
+                m_white_time_elapsed++;
+                update_time_labels(m_time_control_seconds - m_white_time_elapsed, black_time);
+                check_resign_on_time("White time out. Black wins."sv);
+            } else if (m_board.turn() == Chess::Color::Black) {
+                m_black_time_elapsed++;
+                update_time_labels(white_time, m_time_control_seconds - m_black_time_elapsed);
+                check_resign_on_time("Black time out. White wins."sv);
+            }
+        },
+        this);
```

Comment:
This might assume too much about Serenity's capacities at keeping the timer on time :sweat_smile:,
especially since the error will accumulate. Maybe calling `time()` every second and compare that to the time of start will get a more precise result on the long run. But maybe I'm overthinking that, especially if you plan to play bullet lol.
At the same time, if that can :yakbait: you into fixing the rest of the system, that's perfect :smile:

Reply:
I modeled this off of how it was done in the Minesweeper game, which also has a timer. I'll look into using `time()`, if I can't get it working I'll at least add a `FIXME` since this method seems a bit cleaner.